### ISSUE TITLE
Plot point markers and lines in different hues but the same style with seaborn.lineplot

### ISSUE THREAD

**Issue by corytu** [mwaskom/seaborn#1567]
Version information:

- `pandas`: 0.23.4
- `seaborn`: 0.9.0
- `matplotlib`: 2.2.3

Given the dataset and matplotlib backend below:

```python
import pandas as pd
import seaborn as sns
%matplotlib notebook
sns.set()
df = pd.DataFrame({
    "n_index": list(range(5)) * 2,
    "logic": [True] * 5 + [False] * 5,
    "value": list(range(5)) + list(range(5, 10))
})
```

Does not generate markers at all:

- `sns.lineplot(x="n_index", y="value", hue="logic", markers=True, data=df)`

Generates markers but the lines or points are in different styles:

- `sns.lineplot(x="n_index", y="value", hue="logic", style="logic", markers=True, data=df)`
- `sns.lineplot(x="n_index", y="value", hue="logic", style="logic", markers=["o", "o"], data=df)`

Generates markers as desired but the legend ignores what is drown by the artist:

- `sns.lineplot(x="n_index", y="value", hue="logic", marker="o", data=df)`

In sum, the `markers` parameter in [`seaborn.lineplot`](https://seaborn.pydata.org/generated/seaborn.lineplot.html) cannot generate markers in different hues when keeping other elements in the same style, which suggests that two aesthetic dimensions are needed on one data dimension in this use case. I believe that violates the principles of aesthetic mapping. On the other hand, using `marker` parameter from `matplotlib` generates the lines and points as expected, but the legend does not have any marker.

Given the rationale above, I think this is either a needed new feature or a bug need fixing. For more detailed discussion, please visit the question I raised on StackOverflow: [Plot point markers and lines in different hues but the same style with seaborn](https://stackoverflow.com/q/52385428/6666231).

**Comment by mwaskom** [comment]
I saw the StackOverflow question. You got several good answers there. This isn't a bug.

**Comment by corytu** [comment]
@mwaskom Some explanation would be nice. Just because I got good answers doesn't mean there's nothing to be improved.

You said you saw the StackOverflow question, then you should have noticed that the accepted answer was __not__ able to plot the desired output with `seaborn`, and the answerer also admitted that it may be considered as a missing feature or a bug.

The other answer provided a workaround by setting `style` parameter but then turning off `dashes`, and also additionally executed `sns.set_style("darkgrid")`. That does not look like an intuitive way.

**Comment by mwaskom** [comment]
> You said you saw the StackOverflow question, then you should have noticed that the accepted answer was not able to plot the desired output with seaborn

Then perhaps you should have accepted the answer that does ...

**Comment by corytu** [comment]
> Then perhaps you should have accepted the answer that does ...

I've provided the reasons why I didn't accept that, and which answer I chose has nothing to do with this issue. SO is a place to discuss how to make the codes and functionalities _work_, and GitHub is a place to discuss how to make them _better_. Just because it works doesn't mean it can't be better.

Given the fact that the `markers` parameter cannot fulfill the functionality well, I submitted this issue after reading your [CONTRIBUTING.md](https://github.com/mwaskom/seaborn/blob/master/.github/CONTRIBUTING.md).

> Polite requests with an explanation of the proposed feature's virtues will usually get an explanation; feature requests that say "I would like feature X, you need to add it" typically won't.

I am sorry to find that isn't the case even I clearly explained the virtues for aesthetic mapping and data visualization.
